src/Common/XMLUtils.cpp

```diff
@@ -42,15 +42,15 @@ class ParseHelper : private Util::AbstractConfiguration
         return s;
     }
 
-    template <typename ValueType, typename ParseFunction>
+    template <typename ValueType, bool ReturnDefault, typename ParseFunction>
```

Comment:
Maybe it's better to make it the first parameter? `ValueType` and `ParseFunction` can be inferred.

Reply:
Just tried it and got an error:
```
/home/davenger/src/ClickHouse/src/Common/XMLUtils.cpp:74:12: error: no matching function for call to 'getValue'
    return ParseHelper::getValue<false>(node, path, {}, ParseHelper::parseInt64);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/davenger/src/ClickHouse/src/Common/XMLUtils.cpp:46:22: note: candidate template ignored: couldn't infer template argument 'ValueType'

```
For this to work It will require to pass the default value of exact ValueType instead of just '{}'. And that way it will not be any better than the current code.